### PR TITLE
Add a simple js validationscheme

### LINKED ISSUE #43: Validation resizes parent card
Right now, the validation method adds extra content the card. Showing validation tips causes the cards to resize. Is there a way to account for the space in the html, without getting this jittery experience?

### PR DESCRIPTION
**Pull request by andy7809**

### CODE REVIEW

**Comment by andy7809** [comment]
Don't merge yet, #43 needs to be addressed in this PR

**Comment by emmapaczkowski** [comment]
The Changes you have made look good to me.

**Review by emmapaczkowski — APPROVED** [review]

File: where-to-travel/src/main/webapp/signup.js [hunk 1]
```diff
@@ -5,34 +5,53 @@ const passwordSignUp = document.getElementById('passwordSignUp');
...
 
+  const confirmPass = document.getElementById('passwordConfirmation');
+  confirmPass.addEventListener('focusout', function(event) {
+    if (event.target.validity.valid === false) {
+      event.target.parentNode.classList.add('was-validated');
+    }
+  });
+
+  // Fetch all the forms we want to apply custom Bootstrap validation styles to
+  var forms = document.getElementsByClassName('needs-validation');
```
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: var -> const

File: where-to-travel/src/main/webapp/signup.js [hunk 2]
```diff
@@ -5,34 +5,53 @@ const passwordSignUp = document.getElementById('passwordSignUp');
...
+  // Loop over them and prevent submission
+  var validation = Array.prototype.filter.call(forms, function(form) {
```
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: var-> let

File: where-to-travel/src/main/webapp/style.css [hunk 3]
```diff
@@ -98,6 +98,7 @@ html {
   left: 0;
   margin: auto;
   position: absolute;
+  width: fit-content;
```
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/style.css:101]: alphabetize attributes

File: where-to-travel/src/main/webapp/signup.js [hunk 4]
```diff
@@ -5,34 +5,53 @@ const passwordSignUp = document.getElementById('passwordSignUp');
...
-}
 
-passwordSignUp.onchange = validatePassword;
-passwordConfirmation.onkeyup = validatePassword;
+  passwordSignUp.onchange = validatePassword;
+  passwordConfirmation.onkeyup = validatePassword;
+  
+  const email = document.getElementById('emailSignUp');
+  email.addEventListener('focusout', function(event) {
+    if (event.target.validity.valid === false) {
```
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: Might be helpful to have a small comment explaining what this is doing

File: where-to-travel/src/main/webapp/signup.js [hunk 5]
```diff
@@ -5,34 +5,53 @@ const passwordSignUp = document.getElementById('passwordSignUp');
...
+      event.target.parentNode.classList.add('was-validated');
+    }
+  });
+
+  const password = document.getElementById('passwordSignUp');
```
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: could each of these blocks of code for email, password, and confirmPass be refactored into one function where you pass in the document id for each one?

**Review by PriyaVarra — COMMENTED** [review]
Just a couple of small style suggestions. Looks good otherwise!

File: where-to-travel/src/main/webapp/signup.js (hunk 2 above)
Comment (andy7809) [reply where-to-travel/src/main/webapp/signup.js]: Validation doesn't change (I don't think) so I changed it to const in ce27fa9

File: where-to-travel/src/main/webapp/signup.js (hunk 1 above)
Comment (andy7809) [reply where-to-travel/src/main/webapp/signup.js]: fixed in ce27fa9

File: where-to-travel/src/main/webapp/style.css (hunk 3 above)
Comment (andy7809) [reply where-to-travel/src/main/webapp/style.css:101]: good catch, done in 192445c

File: where-to-travel/src/main/webapp/signup.js (hunk 4 above)
Comment (andy7809) [reply where-to-travel/src/main/webapp/signup.js]: see 0bb8559

File: where-to-travel/src/main/webapp/signup.js (hunk 5 above)
Comment (andy7809) [reply where-to-travel/src/main/webapp/signup.js]: Done in 0bb8559

**Comment by andy7809** [comment]
Waiting for a final lgtm from @PriyaVarra or @emmapaczkowski, then I will merge this!

**Review by PriyaVarra — APPROVED** [review]
LGTM!

**Comment by andy7809** [comment]
merging and adding now

**Pull request merged**